### ISSUE TITLE
Grow issue when height is specified

### ISSUE THREAD

**Issue by StephaneDelcroix** [xamarin/flex#25]
here's an html snippet, that renders fine on FF. using X.Flex, the light blue box is slightly higher than the other visible ones, and should not

```
<!DOCTYPE html>
<html>
<head>
<style> 
#main {
    width: 500px;
    height: 600px;
    border: 1px solid #c3c3c3;
    display: flex;
    
    flex-wrap: wrap;
    align-items: flex-start;
    flex-direction: column;
}

#main div {
    width: 250px;
    height: 0px;
}
</style>
</head>
<body>

<div id="main">
  <div style="background-color:coral;flex-grow: 1;flex-basis: 0;"></div>
  <div style="background-color:lightblue;flex-grow: 1;flex-basis: 0; height: 50;"></div>
  <div style="background-color:pink;"></div>
  <div style="background-color:cornflowerblue;flex-grow: 1;flex-basis: 0;"></div>
  <div style="background-color:hotpink;"></div>
</div>

</body>
</html>
```

**Comment by lrz** [comment]
Converted this into a C which indeed fails:

```
void
test_grow7(void)
{
    struct flex_item *root = flex_item_with_size(500, 600);

    struct flex_item *child1 = flex_item_with_size(250, 0);
    flex_item_set_grow(child1, 1);
    flex_item_add(root, child1);

    struct flex_item *child2 = flex_item_with_size(250, 50);
    flex_item_set_grow(child2, 1);
    flex_item_add(root, child2);

    struct flex_item *child3 = flex_item_with_size(250, 0);
    flex_item_add(root, child3);

    struct flex_item *child4 = flex_item_with_size(250, 0);
    flex_item_set_grow(child4, 1);
    flex_item_add(root, child4);

    struct flex_item *child5 = flex_item_with_size(250, 0);
    flex_item_add(root, child5);

    flex_layout(root);

    TEST_FRAME_EQUAL(child1, 0, 0, 250, 200);
    TEST_FRAME_EQUAL(child2, 0, 200, 250, 200);
    TEST_FRAME_EQUAL(child3, 0, 400, 250, 0);
    TEST_FRAME_EQUAL(child4, 0, 400, 250, 200);
    TEST_FRAME_EQUAL(child5, 0, 600, 250, 0);

    flex_item_free(root);
}
```

**Comment by lrz** [comment]
The reason is that we consider the flexible space of the container to be 550 (600 - 50, child2's height), which we then divide by 3 and add to child1, child2 and child4's respective heights, therefore child2 has indeed a bigger height than the others.

We have to change the way we determine the flexible space of the root item in order to fix this issue.

**Comment by lrz** [comment]
Should be fixed by 2ea6dd06adf2ab76190c9e9fb7a572c12590520f
